appholder can't sign auth keys

appholder crashes + shows the following error when provisioning:

```
FATAL EXCEPTION: main
  Process: com.android.identity.wallet.purse, PID: 20168
  java.lang.IllegalStateException: Error creating self signed credential
   at com.android.identity.wallet.document.DocumentManager.createSelfSignedDocument(DocumentManager.kt:71)
   at com.android.identity.wallet.viewmodel.SelfSignedViewModel$createSelfSigned$1$1.invokeSuspend(SelfSignedViewModel.kt:54)
   at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
   at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
   at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
   at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
   at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
   at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
   at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
   at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
   Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@abc4d39, Dispatchers.Main.immediate]
  Caused by: java.lang.ClassCastException: com.android.org.conscrypt.OpenSSLECPrivateKey cannot be cast to org.bouncycastle.jce.interfaces.ECPrivateKey
   at com.android.identity.crypto.EcPrivateKeyKt.toEcPrivateKey(EcPrivateKey.kt:144)
   at com.android.identity.wallet.util.ProvisioningUtil.refreshAuthKeys(ProvisioningUtil.kt:193)
   at com.android.identity.wallet.util.ProvisioningUtil.provisionSelfSigned(ProvisioningUtil.kt:85)
   at com.android.identity.wallet.document.DocumentManager.provisionSelfSignedDocument(DocumentManager.kt:159)
   at com.android.identity.wallet.document.DocumentManager.createSelfSignedDocument(DocumentManager.kt:69)
   ... 9 more
```

--- a/appholder/src/main/java/com/android/identity/wallet/document/KeysAndCertificates.kt
+++ b/appholder/src/main/java/com/android/identity/wallet/document/KeysAndCertificates.kt
@@ -2,6 +2,7 @@ package com.android.identity.wallet.document
 
 import android.content.Context
 import com.android.identity.wallet.R
+import org.bouncycastle.jce.provider.BouncyCastleProvider
 import java.io.InputStream
 import java.nio.charset.StandardCharsets
 import java.security.KeyFactory
@@ -38,14 +39,14 @@ object KeysAndCertificates {
     private fun getPrivateKey(context: Context, resourceId: Int): PrivateKey {
         val keyBytes: ByteArray = getKeyBytes(context.resources.openRawResource(resourceId))
         val spec = PKCS8EncodedKeySpec(keyBytes)
-        val kf = KeyFactory.getInstance("EC")
+        val kf = KeyFactory.getInstance("EC", BouncyCastleProvider.PROVIDER_NAME)
         return kf.generatePrivate(spec)
     }
 
     private fun getPublicKey(context: Context, resourceId: Int): PublicKey {
         val keyBytes: ByteArray = getKeyBytes(context.resources.openRawResource(resourceId))
         val spec = X509EncodedKeySpec(keyBytes)
-        val kf = KeyFactory.getInstance("EC")
+        val kf = KeyFactory.getInstance("EC", BouncyCastleProvider.PROVIDER_NAME)
         return kf.generatePublic(spec)
     }
 

